routes/extra_data_utils.go

```diff
@@ -0,0 +1,38 @@
+package routes
+
+import (
+	"encoding/hex"
+	"github.com/deso-protocol/core/lib"
+	"strconv"
+)
+
+type DecoderFunc func([]byte) string
+
+type ExtraDataDecoder struct {
+	Key     string
+	Decoder DecoderFunc
+}
+
+// GetExtraDataDecoderFunc Values in ExtraData field
+// in transaction may have special encoding. In such cases
+// we'll need specialized decoders too
+func GetExtraDataDecoderFunc(key string) DecoderFunc {
+	var decoders = []ExtraDataDecoder{
+		{
+			lib.DiamondLevelKey,
+			ByteArrTo64BitInt,
+		},
```

Comment:
We should add decoders for the other consensus keys. You can find them listed here: https://github.com/deso-protocol/core/blob/1b9d4d61499d66f3c3f2911496785aa6b95dd5d5/lib/constants.go#L846.

Also maybe we converted ExtraData from a map of string to string into a map of string to interface

Reply:
Yeah, ExtraData should ultimately be a map of string to interface. Some ExtraData values themselves are maps, so we'll need to support some hierarchy. Best to make this change in a separate PR